common/utils/helpers.py

```
@@ -0,0 +1,38 @@
+def _check_ownership(self, obj, user):
+    """Helper method to verify object ownership."""
+    if obj.added_by != user:
+        raise PermissionDenied(
+            f"Object {obj.id} does not belong to user {user.username}"
+        )
+
+
+class BaseTestCase(TestCase):
+    """Base test class with common setup and helper methods."""
+
+    def setUp(self):
```

Ruff (0.12.2)

Missing return type annotation for public function `setUp`

Add return type annotation: `None`

(ANN201)